app/services/npq/create_participant_outcome.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module NPQ
+  class CreateParticipantOutcome
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :cpd_lead_provider
+    attribute :participant_external_id
+    attribute :course_identifier
+    attribute :state
+    attribute :completion_date, :date
+
+    validates :cpd_lead_provider, :participant_external_id, presence: true
+    validates :completion_date, future_date: true, presence: { message: I18n.t(:missing_completion_date) }
+    validates :course_identifier, course: true, presence: { message: I18n.t(:missing_course_identifier) }
+    validates :state,
+              presence: { message: I18n.t(:missing_state) },
+              inclusion: {
+                in: ParticipantOutcome::NPQ::VALID_STATES.map(&:to_s),
+                message: I18n.t(:invalid_state),
+              }
+    validate :participant_profile_has_no_completed_declarations
+
+    def call
+      return if invalid?
+
+      if participant_outcome_already_exists?
+        latest_participant_outcome
+      else
+        new_participant_outcome.save!
+        participant_declaration.outcomes.latest
+      end
+    end
+
+    def participant_identity
+      @participant_identity ||= ParticipantIdentity.find_by(external_identifier: participant_external_id)
+    end
+
+  private
+
+    def new_participant_outcome
+      @new_participant_outcome ||= ParticipantOutcome::NPQ.new(participant_declaration:, state:, completion_date:)
+    end
+
+    def latest_participant_outcome
+      @latest_participant_outcome ||= participant_declaration&.outcomes&.latest
+    end
+
+    def participant_outcome_already_exists?
+      return false if latest_participant_outcome.blank?
+
+      latest_participant_outcome.slice(:state, :completion_date) == new_participant_outcome.slice(:state, :completion_date)
+    end
+
+    def participant_profile
+      @participant_profile ||= ParticipantProfileResolver
+                                .call(
+                                  participant_identity:,
+                                  course_identifier:,
+                                  cpd_lead_provider:,
+                                )
+    end
+
+    def participant_declarations
+      return unless participant_profile
+
+      @participant_declarations ||= participant_profile.participant_declarations.npq.for_declaration("completed").order(declaration_date: :desc)
+    end
+
+    def participant_profile_has_no_completed_declarations
+      return unless participant_profile
```

Comment:
I'm a bit confused about this - surely we should check the declaration we are attaching this outcome to is `completed`

Reply:
Hi @peteglondon yeah I agree and we're selecting only the completed ones for the validation on https://github.com/DFE-Digital/early-careers-framework/pull/2809/files#diff-367ec6ccda5399216a9ddb942998745f3dbe278e4591b2a37ac28a6520bfcc0aR68